core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -63,7 +65,7 @@ private[spark] object PythonEvalType {
  */
 private[spark] abstract class BasePythonRunner[IN, OUT](
     funcs: Seq[ChainedPythonFunctions],
-    evalType: Int,
+    evalTypes: Seq[Int],
```

Comment:
So you have a Seq now to support a mix of unbounded/bounded eval types?  Is there any other way to do this, like maybe decipher from the bounds given?

Reply:
There might be other way to do this. I like extending a single eval type to a seq of eval types because it seems pretty simple and flexible to me.

Reply:
The only eval types that can be used together are unbounded and bounded windows right? I only worry that this might lead to more complications with the other types that cannot be sent to the worker together.

Reply:
I see your point - I can see this being used for other things too, for example, numpy variant vectorized UDFs, or Window transform UDFs for unbounded window (n -> n mapping for unbounded window, such as rank). I choose this approach because of the flexibility.

For this particular case, it is possible to distinguish between bounded/unbounded, for example, maybe sending something in the arg offsets or sth like that, but this would be using arg offsets for sth else...

Reply:
I don't see that the additional complexity this adds is worth it for now, but curious what others think. 

If I understand correctly, the python worker just takes an index range for bounded windows and the entire range for unbounded. It does not really care about anything else. So couldn't you just send an index that encompasses the entire range for unbounded? Then you would only need to define `SQL_WINDOW_AGG_PANDAS_UDF` in the worker and all the same code for both, which would simplify quite a bit.

Reply:
> So couldn't you just send an index that encompasses the entire range for unbounded

This is actually what I first did. However, I think this would require sending more data than necessary for the unbounded case. In worst case it will be 3x the number of columns (begin_index, end_index, data) comparing to just one column (data).

Reply:
Could you just send a flag to indicate it has bounds or not? or add something to `runner_conf`?

Reply:
Yea, setting `evalTypes` to `Seq` doesn't looks a great idea. I agree with https://github.com/apache/spark/pull/22305/files#r223774544.

Reply:
If that's going to require another necessary change, then please go ahead. At least two committers here understood why it's introduced here.

Reply:
I have reverted the evalType change and use a runner config instead.